### PR TITLE
1017/unmanaged ios

### PR DESCRIPTION
**Pull request by commjoen**
Thank you for submitting a Pull Request to the Mobile Security Testing Guide. Please make sure that:

- [x] Your contribution is written in the 2nd person (e.g. you)
- [x] Your contribution is written in an active present form for as much as possible.
- [x] You have made sure that the reference section is up to date (e.g. please add sources you have used, make sure that the references to MITRE/MASVS/etc. are up to date)
- [x] Your contribution has proper formatted markdown and/or code
- [x] Any references to website have been formatted as [TEXT](URL “NAME”)
- [x] You verified/tested the effectiveness of your contribution (e.g.: is the code really an effective remediation? Please verify it works!)

If your PR is related to an issue. Please end your PR test with the following line:
This PR covers issue #1017.

### CODE REVIEW

File: Document/0x06i-Testing-Code-Quality-and-Build-Settings.md [hunk 1]
```diff
@@ -491,10 +491,39 @@ However, these do not include all the libraries being used. Next, with Class-dum
...
+- Retaining cycles: look for cyclic dependencies by means of strong references of components to one another which keep materials in memory.
+- Using instances of `UnsafePointer` can be managed wrongly, which will allow for various memory corruption issues.
+- Trying to manage the reference count to an object by `Unmanaged` manually, leading to wrong counter numbers and a too late/too soon release.
+
+[A great talk is given on this subject at Realm academy](https://academy.realm.io/posts/russ-bishop-unsafe-swift/ "Russh Bishop on Unsafe Swift") and [a nice tutorial to see what is actually happening](https://www.raywenderlich.com/780-unsafe-swift-using-pointers-and-interacting-with-c "Unsafe Swift: Using Pointers And Interacting With C") is provided by Ray Wenderlich on this subject.
+
+>Please note that with Swift 5 you can only deallocate full blocks, which means the playground has changed a bit.
+
+#### Dynamic Analysis
+There are various tools provided which which memory bugs can be found:
```
Comment (sushi2k) [inline Document/0x06i-Testing-Code-Quality-and-Build-Settings.md]: There are various tools provided which which memory bugs can be found:
-> There are various tools provided which help to identify memory bugs:

File: Document/0x06i-Testing-Code-Quality-and-Build-Settings.md [hunk 2]
```diff
@@ -491,10 +491,39 @@ However, these do not include all the libraries being used. Next, with Class-dum
...
+- Apple its Debug Memory graph introduced in XCode 8.
+- The Allocations and Leaks instrument in XCode.
```
Comment (sushi2k) [inline Document/0x06i-Testing-Code-Quality-and-Build-Settings.md]: This is Instruments in Xcode, right? Then just the following should be sufficient:
- Instruments as part of Xcode

File: Document/0x06i-Testing-Code-Quality-and-Build-Settings.md [hunk 3]
```diff
@@ -491,10 +491,39 @@ However, these do not include all the libraries being used. Next, with Class-dum
...
+- Check whether memory is freed too fast or to slow by enabling `NSAutoreleaseFreedObjectCheckEnabled`, `NSZombieEnabled`, `NSDebugEnabled` in Xcode while testing the application.
```
Comment (sushi2k) [inline Document/0x06i-Testing-Code-Quality-and-Build-Settings.md]: As this is not really a tool anymore, start the sentence different:

Besides the tools listed above, you can also check whether memory is freed too fast or too slow by enabling `NSAutoreleaseFreedObjectCheckEnabled`, `NSZombieEnabled`, `NSDebugEnabled` in Xcode while testing the application.

**Comment by commjoen** [comment]
fixed all feedback @TheDauntless & @sushi2k : can you check again?

File: Document/0x06i-Testing-Code-Quality-and-Build-Settings.md [hunk 4]
```diff
@@ -491,10 +491,38 @@ However, these do not include all the libraries being used. Next, with Class-dum
...
+- Retaining cycles: look for cyclic dependencies by means of strong references of components to one another which keep materials in memory.
+- Using instances of `UnsafePointer` can be managed wrongly, which will allow for various memory corruption issues.
+- Trying to manage the reference count to an object by `Unmanaged` manually, leading to wrong counter numbers and a too late/too soon release.
+
+[A great talk is given on this subject at Realm academy](https://academy.realm.io/posts/russ-bishop-unsafe-swift/ "Russh Bishop on Unsafe Swift") and [a nice tutorial to see what is actually happening](https://www.raywenderlich.com/780-unsafe-swift-using-pointers-and-interacting-with-c "Unsafe Swift: Using Pointers And Interacting With C") is provided by Ray Wenderlich on this subject.
+
+>Please note that with Swift 5 you can only deallocate full blocks, which means the playground has changed a bit.
+
+#### Dynamic Analysis
+There are various tools provided which help to identify memory bugs within Xcode, such as the Debug Memory graph introduced in XCode 8 and the Allocations and Leaks instrument in XCode.
```
Comment (sushi2k) [inline Document/0x06i-Testing-Code-Quality-and-Build-Settings.md]: Please use "Xcode" consistently

File: Document/0x06i-Testing-Code-Quality-and-Build-Settings.md [hunk 5]
```diff
@@ -491,10 +491,38 @@ However, these do not include all the libraries being used. Next, with Class-dum
...
+
+Next, you can check whether memory is freed too fast or to slow by enabling `NSAutoreleaseFreedObjectCheckEnabled`, `NSZombieEnabled`, `NSDebugEnabled` in Xcode while testing the application.
```
Comment (sushi2k) [inline Document/0x06i-Testing-Code-Quality-and-Build-Settings.md]: to slow -> too slow

**Comment by OMTGreleaser** [comment]
Results for commit 68a5641cb7fbd050d9b161148298bb796368a398: Broken link result: [!] Error(s) found by the Links validator: 1 pages have dead links! Verbose output in /link-check-result.out, markdown result errorlength:        0 ../linter-result.out. If fields are empty or 0: you should be OK. If fields contain data: please check for markdown and link errors.

**Comment by OMTGreleaser** [comment]
Results for commit e40b593edf934baf0d508be30ef54bc817c8382b: Broken link result: [!] Error(s) found by the Links validator: 1 pages have dead links! Verbose output in /link-check-result.out, markdown result errorlength:        0 ../linter-result.out. If fields are empty or 0: you should be OK. If fields contain data: please check for markdown and link errors.

**Comment by OMTGreleaser** [comment]
Results for commit da99e76d9862e416ff09631b18a98f3f29b9cfde: Broken link result: [+] No error found by the Links validator., markdown result errorlength:        0 ../linter-result.out. If fields are empty or 0: you should be OK. If fields contain data: please check for markdown and link errors.

**Comment by commjoen** [comment]
Fixed all the feedback...

**Comment by OMTGreleaser** [comment]
Results for commit 58b113fc51f1126079687169aee9e0fa0f7d91d4: Broken link result: [!] Error(s) found by the Links validator: 2 pages have dead links! Verbose output in /link-check-result.out, markdown result errorlength:        0 ../linter-result.out. If fields are empty or 0: you should be OK. If fields contain data: please check for markdown and link errors.

**Comment by OMTGreleaser** [comment]
Results for commit 58b113fc51f1126079687169aee9e0fa0f7d91d4: Broken link result: [!] Error(s) found by the Links validator: 1 pages have dead links! Verbose output in /link-check-result.out, markdown result errorlength:        0 ../linter-result.out. If fields are empty or 0: you should be OK. If fields contain data: please check for markdown and link errors.

**Comment by OMTGreleaser** [comment]
Results for commit 3fbabade82ff9bd41b65a0d7d5599bf2226e903a: Broken link result: [+] No error found by the Links validator., markdown result errorlength:        0 ../linter-result.out. If fields are empty or 0: you should be OK. If fields contain data: please check for markdown and link errors.

**Pull request merged**